### PR TITLE
Feature/1558/1559/1649

### PR DESCRIPTION
**Pull request by Kacper-RF**
### Pull Request Checklist

- [ ] I have linked this PR to related GitHub issue.
- [x] I have updated the CHANGELOG.md file with relevant changes (the file is located at the root of monorepo).

Only desktop part at this moment

https://github.com/TryQuiet/quiet/issues/1558
https://github.com/TryQuiet/quiet/issues/1559
https://github.com/TryQuiet/quiet/issues/1649

### CODE REVIEW

File: packages/desktop/src/renderer/components/widgets/channels/ChannelMessages.tsx [hunk 1]
```diff
@@ -152,6 +153,7 @@ export const ChannelMessagesComponent: React.FC<IChannelMessagesProps & FileActi
                     downloadFile={downloadFile}
                     cancelDownload={cancelDownload}
                     onMathMessageRendered={onMathMessageRendered}
+                    // place for userLabel props
```
Comment (siepra) [inline packages/desktop/src/renderer/components/widgets/channels/ChannelMessages.tsx:156]: Is this comment neccessary?

File: packages/desktop/src/renderer/components/widgets/channels/BasicMessage.tsx [hunk 2]
```diff
@@ -207,6 +210,11 @@ export const BasicMessageComponent: React.FC<BasicMessageProps & FileActionsProp
                       {infoMessage ? 'Quiet' : messageDisplayData.nickname}
                     </Typography>
                   </Grid>
+                  {userLabel && (
+                    <Grid>
+                      <UserLabelContainer type={userLabel} username={messageDisplayData.nickname} />
```
Comment (siepra) [inline packages/desktop/src/renderer/components/widgets/channels/BasicMessage.tsx:215]: Why did you add "container" suffix?

Comment (siepra) [reply packages/desktop/src/renderer/components/widgets/channels/BasicMessage.tsx:215]: Ok, I see. Won't it break storybook/snapshot testing? I guess there should probably not be a separate container for user label only.

File: packages/desktop/src/renderer/sagas/modals/modals.types.ts [hunk 3]
```diff
@@ -21,4 +21,6 @@ export enum ModalName {
   searchChannelModal = 'searchChannelModal',
   warningModal = 'warningModal',
   channelCreationModal = 'channelCreationModal',
+  duplicateModal = 'duplicateModal',
```
Comment (siepra) [inline packages/desktop/src/renderer/sagas/modals/modals.types.ts:24]: I'm missing the information what specifically is duplicated. I think naming it `duplicatedUserModal` or `duplicatedUsernameModal` would be better

File: packages/desktop/src/renderer/sagas/modals/modals.types.ts [hunk 4]
```diff
@@ -21,4 +21,6 @@ export enum ModalName {
...
+  unregisteredModal = 'unregisteredModal',
```
Comment (siepra) [inline packages/desktop/src/renderer/sagas/modals/modals.types.ts:25]: Same as above

File: packages/desktop/src/renderer/components/widgets/userLabel/UserLabel.component.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+const UserLabel: React.FC<UserLabelProps> = ({
+  type,
+  username,
+  unregisteredUsernameModalHandleOpen,
+  duplicatedUsernameModalHandleOpen,
+}) => {
+  const isUnregistered = type === UserLabelType.UNREGISTERED
+  const handleOpen = isUnregistered ? unregisteredUsernameModalHandleOpen : duplicatedUsernameModalHandleOpen
```
Comment (EmiM) [inline packages/desktop/src/renderer/components/widgets/userLabel/UserLabel.component.tsx:61]: How does it look when user is both unregistered and has duplicated username?

Comment (Kacper-RF) [reply packages/desktop/src/renderer/components/widgets/userLabel/UserLabel.component.tsx:61]: I built it to provide one label (just like on Figma), we can decide what is more important in that situation and display that label as quick solution.

For long term solution in next iteration we can work on possibility to show 2 different labels

**Review by siepra — APPROVED** [review]

**Review by EmiM — APPROVED** [review]
